### PR TITLE
[TC-834] mail notification module

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[TC-0834](https://bcdevex.atlassian.net/browse/TC-0834)

Objective:

- includes generic email svc
- includes supervisor initiated triggers <-- can be tested by setting members status to MEMBER_COMMITTED and then visiting supervisor dash to approve/decline members
- TODO: content of inner templates
- TODO: member initiated triggers
- TODO: scheduled reminder triggers

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -29,4 +29,9 @@ CHEFS_WS_TOKEN=
 CHIPS_API=https://analytics-testapi.psa.gov.bc.ca/apiserver/api.rsc/Datamart_COREProg_dbo_vw_report_CoreProg_EmployeeData
 CHIPS_API_KEY=
 
-BCWS_TOKEN=
\ No newline at end of file
+BCWS_TOKEN=
+
+CHESS_API='https://ches-dev.api.gov.bc.ca/api/v1'
```
Comment (fw-noel) [inline .env.example:34]: Just quickly, shouldn't this be `CHES`?

File: backend/src/mail/mail.service.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  async sendMail(body: MailDto) {
+    const token = await this.getToken();
+
+    this.mailApi.defaults.headers.common['Authorization'] = `Bearer ${token}`;
+
+    try {
+      const res = await this.mailApi.post('/emailMerge', body);
+      return res.data;
+    } catch (e) {
+      console.log('ERROR');
```
Comment (fw-noel) [inline backend/src/mail/mail.service.ts:91]: console.error?

File: backend/src/mail/utils.ts [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+  });
+};
+
+export const renderTemplate = (
+  template: EmailTemplates,
+  personnel: PersonnelEntity[],
+  program?: Program,
+) => {
+  let baseUrl;
+  if (process.env.NODE_ENV === 'production') {
```
Comment (fw-noel) [inline backend/src/mail/utils.ts:122]: Should this be moved to an ENV?

File: backend/src/mail/views/blocks/member-button.njk [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<div class="button-div">
+  <a id="link-button" href="https://coreteam.gov.bc.ca/profile">Get Started</a>
```
Comment (fw-noel) [inline backend/src/mail/views/blocks/member-button.njk:2]: Given that there's a link to production here, can we ensure that mail from TEST / DEV are CLEARLY marked as such?

**Review by fw-noel — DISMISSED** [review]

**Review by fw-noel — DISMISSED** [review]

**Review by fw-noel — APPROVED** [review]

**Comment by chelsea-EYDS** [comment]
prev approved

**Pull request merged**